synapse/app/generic_worker.py

```diff
@@ -459,12 +464,7 @@ def start(config_options: List[str]) -> None:
 
     if config.worker.worker_app == "synapse.app.user_dir":
         if not config.worker.update_user_directory:
-            sys.stderr.write(
-                "\nThe user directory worker is not allowed to perform user directory updates per the config."
-                "\nPlease add or edit "
-                f'``update_user_directory_on: "{config.worker.worker_name}"`` in the config\n'
-            )
-            sys.exit(1)
+            logger.warning(DIRECTORY_UPDATE_WARNING.format(config.worker.worker_name))
```

Comment:
given that this happens before logging is set up, I think it probably just ends up in stderr anyway. It may be best to stick to `sys.stderr.write` for consistency.